dynatrace/src/test/java/com/dynatrace/opentelemetry/metric/OneAgentMetadataEnricherTest.java

```diff
@@ -0,0 +1,42 @@
+package com.dynatrace.opentelemetry.metric;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.logging.Logger;
+import org.junit.jupiter.api.Test;
+
+public class OneAgentMetadataEnricherTest {
```

Comment:
Please add tests to verify proper handling of missing/inaccessible/invalid indirection and metadata files.